### PR TITLE
Purge old monitor requests from database

### PR DESCRIPTION
**Pull request by karlb**
Relates to https://github.com/raiden-network/raiden-services/issues/419.

### CODE REVIEW

**Comment by karlb** [comment]
The failure in CI could be due to different sqlite versions. I'm looking into it.

File: src/monitoring_service/schema.sql [hunk 1]
```diff
@@ -48,10 +48,20 @@ CREATE TABLE monitor_request (
     reward_proof_signature  CHAR(132)   NOT NULL,
 
     non_closing_signer      CHAR(42)    NOT NULL,
+
+    -- These two columns are just for handling MRs before we have confirmed
+    -- that a matching channel exists.
+    -- * If `waiting_for_channel` is false, we've already checked that such a
+    --   channel exists and everything is ok.
+    -- * If `saved_at` is sufficiently recent, a missing channel is acceptable.
+    -- * If `saved_at` is old, we will delete the MR if not matching channel is
```
Comment (palango) [inline src/monitoring_service/schema.sql]: s/not/no

File: src/monitoring_service/schema.sql [hunk 2]
```diff
@@ -48,10 +48,20 @@ CREATE TABLE monitor_request (
...
+    --   found.
+    saved_at                TIMESTAMP   NOT NULL DEFAULT CURRENT_TIMESTAMP,
```
Comment (palango) [inline src/monitoring_service/schema.sql]: Maybe add a comment that `CURRENT_TIMESTAMP` is GMT by default.

File: src/monitoring_service/service.py [hunk 3]
```diff
@@ -186,3 +188,32 @@ def _process_new_blocks(self, last_block: BlockNumber) -> None:
...
+                WHERE waiting_for_channel
+                  AND EXISTS (
+                    SELECT 1
+                    FROM channel
+                    WHERE channel.identifier = monitor_request.channel_identifier
+                      AND channel.token_network_address = monitor_request.token_network_address
+                  )
+            """
+            )
+            before_this_is_old = datetime.utcnow() - timedelta(minutes=15)
```
Comment (palango) [inline src/monitoring_service/service.py]: In other places I needed to use `datetime.now(timezone.utc)` to get proper timezone enabled datetimes, should we do that here as well?

Comment (palango) [inline src/monitoring_service/service.py]: I'd make this a constant, so it is more discoverable.

File: tests/monitoring/fixtures/factories.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            chain_id=chain_id,
+            nonce=nonce,
+            additional_hash="",
+            balance_hash=encode_hex(bytes([amount])),
+            priv_key=get_random_privkey(),
+        )
+        request_monitoring = balance_proof.get_request_monitoring(
+            privkey=non_closing_privkey,
+            reward_amount=TokenAmount(55),
+            monitoring_service_contract_address=Address(bytes([11] * 20)),
```
Comment (palango) [inline tests/monitoring/fixtures/factories.py]: Maybe use `TEST_MSC_ADDRESS` here

File: src/monitoring_service/service.py (hunk 3 above)
Comment (karlb) [reply src/monitoring_service/service.py]: Both are without timezone, so we should be fine. All the trouble we had was just related to marshmallow adding a time zone, which caused problems for equality checks and signatures, right? I don't remember the time actually being off in any case.

Comment (palango) [reply src/monitoring_service/service.py]: That's true, my question was more if we should care to use this consistently or not.

**Pull request merged**